### PR TITLE
speedup alertlog widget

### PR DESCRIPTION
**Pull request by Negatifff**
Rework main sql request for widget 'alert history' - the query to the table was placed in a separate query and the processing of the linkage of the main query with this table on the php level, which made it possible to speed up the work of the widget by at least 15 times.

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by murrant** [comment]
If you are going to rewrite SQL, please use Laravel Eloquent/Fluent. Otherwise it will need to be rewritten again later anyway.

Also, scopes might be of interest to you.

**Comment by Negatifff** [comment]
> If you are going to rewrite SQL, please use Laravel Eloquent/Fluent. Otherwise it will need to be rewritten again later anyway.
> 
> Also, scopes might be of interest to you.

unfortunately my knowledge of librenms and laravel framework is not enough to make such changes. maybe someday ...

File: includes/html/table/alertlog.inc.php [hunk 1]
```diff
@@ -13,15 +13,16 @@
  * @author     LibreNMS Contributors
 */
 
-$alert_severities = [
+$alert_severities = array(
     // alert_rules.status is enum('ok','warning','critical')
     'ok' => 1,
     'warning' => 2,
     'critical' => 3,
```
Comment (SourceDoctor) [inline includes/html/table/alertlog.inc.php:20]: moving this Array to an Enum would be appreciated
take a look on LibreNMS\Enum directory for examples

File: includes/html/table/alertlog.inc.php [hunk 2]
```diff
@@ -13,15 +13,16 @@
  * @author     LibreNMS Contributors
 */
 
-$alert_severities = [
+$alert_severities = array(
```
Comment (SourceDoctor) [inline includes/html/table/alertlog.inc.php:16]: please change "array()" back to "[.,..]"
"array()" is old style

Comment (Negatifff) [reply includes/html/table/alertlog.inc.php]: done

**Comment by Jellyfrog** [comment]
Thank you for your contribution but we're closing this due to inactivity.  
Feel free to reopen if you plan on finishing it!

**Comment by Negatifff** [comment]
> Thank you for your contribution but we're closing this due to inactivity.
> Feel free to reopen if you plan on finishing it!

this is a working fix for the existing widget variant. but I unfortunately cannot rewrite it the way murrant asked to do it - using Laravel Eloquent / Fluent

**Pull request closed without merging**